openlibrary/plugins/openlibrary/js/SearchPage.js

```diff
@@ -0,0 +1,34 @@
+import * as SearchUtils from './SearchUtils';
+
+export class SearchPage {
+    /**
+     * @param {SearchState} searchState
+     */
+    constructor(searchState) {
+        this.searchState = searchState;
+        this.searchState.sync('searchMode', () => SearchUtils.updateSearchMode('.olform', this.searchState.searchMode));
+
+        // updateWorkAvailability is defined in openlibrary\openlibrary\plugins\openlibrary\js\availability.js
+        // eslint-disable-next-line no-undef
+        updateWorkAvailability();
+
+        $('.search-mode').change(event => {
```

Comment:
This is a class that impacts global state. I'd much prefer it if this was a function that took $('.search-mode') as a parameter - it would make behaviour must more predictable and the code more reusable.

Reply:
Haha, great minds think alike :P See the future iterations.